esmvaltool/diag_scripts/land_carbon_cycle/diag_zonal_turnover.py

```diff
@@ -0,0 +1,226 @@
+                     lw=0.5,
+                     label=row_mod)
+
+    leg = plut.draw_line_legend(ax_fs=fig_config['ax_fs'])
+
+    plt.gca().set_xscale('log')
+    plt.xlim(fig_config['valrange_x'][0], fig_config['valrange_x'][1])
+    plt.ylim(fig_config['valrange_y'][0], fig_config['valrange_y'][1])
+    plt.axhline(y=0, lw=0.48, color='grey')
+    plt.xlabel(f'{tau_obs.long_name} ({tau_obs.units})',
+               fontsize=fig_config['ax_fs'],
+               ma='center')
+    plt.ylabel(f'{lats_obs.long_name} ({lats_obs.units})',
+               fontsize=fig_config['ax_fs'],
+               ma='center')
+    plut.rem_ax_line(['top', 'right'])
+
+    plut.save_figure(plot_path, _extr_art=[leg])
+    plt.close()
+
+
+def main(diag_config):
+    """
+    Diagnostic function to compare the zonal turnover time.
+
+    Argument:
+    --------
+        diag_config - nested dictionary of metadata
+    """
+    my_files_dict = group_metadata(diag_config['input_data'].values(),
+                                   'dataset')
+
+    fig_config = _get_fig_config(diag_config)
+    zonal_tau_mod = {}
+    for key, value in my_files_dict.items():
+        zonal_tau_mod[key] = {}
+        ctotal = _load_variable(value, 'ctotal')
+        gpp = _load_variable(value, 'gpp')
+        zonal_tau_mod[key] = _calc_zonal_tau(gpp, ctotal, fig_config)
+
+    zonal_tau_obs = _get_obs_data_zonal(diag_config)
+
+    obs_var = diag_config.get('obs_variable')[0]
+    tau_obs = zonal_tau_obs[obs_var]
+    base_name = '{title}_{source_label}_{grid_label}z'.format(
+        title=tau_obs.long_name,
+        source_label=diag_config['obs_info']['source_label'],
+        grid_label=diag_config['obs_info']['grid_label'])
+
+    provenance_record = get_provenance_record(
+        "Comparison of latitudinal (zonal) variations of observation-based and"
+        "modelled ecosystem carbon turnover time. The zonal turnover time is"
+        "calculated as the ratio of zonal `ctotal` and `gpp`. Reproduces "
+        "figure 2a and 2b in Carvalhais et al. (2014).", ['mean', 'perc'],
+        ['zonal'], _get_ancestor_files(diag_config, obs_var))
+
+    if diag_config['write_netcdf']:
+        model_cubes = [
+            c for c in zonal_tau_mod.values() if isinstance(c, iris.cube.Cube)
+        ]
+        obs_cubes = [
+            c for c in zonal_tau_obs.values() if isinstance(c, iris.cube.Cube)
+        ]
+        netcdf_path = get_diagnostic_filename(base_name, diag_config)
+        save_cubes = iris.cube.CubeList(model_cubes + obs_cubes)
+        iris.save(save_cubes, netcdf_path)
+    else:
+        netcdf_path = None
+
+    if diag_config['write_plots']:
+        plot_path = get_plot_filename(base_name, diag_config)
+        _plot_zonal_tau(plot_path, zonal_tau_mod, zonal_tau_obs, diag_config)
+        provenance_record['plot_file'] = plot_path
+
+    with ProvenanceLogger(diag_config) as provenance_logger:
+        provenance_logger.log(netcdf_path, provenance_record)
```

Comment:
```suggestion
        with ProvenanceLogger(diag_config) as provenance_logger:
            provenance_logger.log(plot_path, provenance_record)
```
would work better in case `netcdf_path` is not defined. If I recall correctly, there is also a bug with using the `plot_file` key of the provenance record to include provenance in the figure, I should probably remove the feature or fix the bug or update the example sometime soon.

Reply:
Changed accordingly.